src/uportSubprovider.js

```diff
@@ -22,8 +23,9 @@ class UportSubprovider {
       if (self.address) return cb(null, self.address)
       requestAddress().then(
         address => {
-          self.address = address
-          cb(null, address)
+          // TODO consider throwing warning if MNID does not match set network
```

Comment:
@zachferland This is a really good idea. Do the callback with an error explaining what is going on if the address doesn't match the network.